tests/integration/devfile/cmd_devfile_describe_test.go

```diff
@@ -107,4 +127,66 @@ var _ = Describe("odo devfile describe command tests", func() {
 		})
 	})
 
+	Context("devfile has missing metadata", func() {
+		// Reference: https://github.com/openshift/odo/issues/4815
+		var metadata devfilepkg.DevfileMetadata
+
+		// copyAndCreate copies required source code and devfile, and creates a component
+		var copyAndCreate = func(path string) {
+			// Using SpringBoot example here because it helps to distinguish between language and projectType.
+			// In terms of SpringBoot, spring in the projectType and java is the language
+			helper.CopyExample(filepath.Join("source", "devfiles", "springboot", "project"), commonVar.Context)
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "springboot", path), filepath.Join(commonVar.Context, "devfile.yaml"))
+			helper.Cmd("odo", "create", "--context", commonVar.Context).ShouldPass()
+		}
+
+		// checkDescribe checks the describe output (both normal and json) to see if it contains the expected componentType
+		var checkDescribe = func(componentType string) {
+			By("checking the normal output", func() {
```

Comment:
What's the purpose of `By`? I'm asking because I expected something nested in it but I didn't find any `It` clause in it.

Reply:
Quoting from https://onsi.github.io/ginkgo/#documenting-complex-its-by
>As a rule, you should try to keep your Its, BeforeEaches, etc. short and to the point. Sometimes this is not possible, particularly when testing complex workflows in integration-style tests. In these cases your test blocks begin to hide a narrative that is hard to glean by looking at code alone. Ginkgo provides by to help in these situations.

`By` is a way of structuring checks inside the `It`.

Reply:
Thanks.